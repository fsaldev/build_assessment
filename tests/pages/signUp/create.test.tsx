import * as React from 'react'
import { shallow, mount } from 'enzyme';
import Create from '../../../pages/signup/create';
import wrapper from '../../../store'

it('should render correctly with no props', () => {
  const SignUpWrapper = wrapper.withRedux(Create)
  const component = mount(<SignUpWrapper/>)
  
  
  expect(component).toMatchSnapshot();
});
