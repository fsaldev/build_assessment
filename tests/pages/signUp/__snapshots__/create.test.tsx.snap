// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with no props 1`] = `
<withRedux(CreateNEAR)>
  <Provider
    store={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <CreateNEAR>
      <div
        className="flex flex-col items-center justify-center"
      >
        <HomeHeaderText
          Text="Create NEAR account"
        >
          <Header
            Title={[Function]}
          >
            <div
              className="w-full flex items-center justify-center relative bg-white-smoke py-4"
            >
              <title>
                <span
                  className="text-black font-medium text-xl"
                >
                  Create NEAR account
                </span>
              </title>
            </div>
          </Header>
        </HomeHeaderText>
        <div
          className=" px-6 flex flex-row items-center justify-center mt-6"
        >
          <p
            className="text-start text-gray-500"
          >
            Enter an Account ID to use with your NEAR account. Your Account ID will be used for all NEAR operations, including sending and receiving assets.
          </p>
        </div>
        <div
          className="mt-8 w-full px-6 "
        >
          <p
            className="text-start text-gray-500 "
          >
            Full Name
          </p>
          <InputField
            onChange={[Function]}
            placeholder="Ex. John Doe"
            type="name"
            value=""
          >
            <div
              className="mb-6 rounded-md"
            >
              <input
                className="rounded-lg bg-gray-50 appearance-none border-2 border-gray-300 w-full py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-accent-theme"
                onChange={[Function]}
                placeholder="Ex. John Doe"
                type="name"
                value=""
              />
            </div>
          </InputField>
          <div
            className="flex flex-row items-start justify-start"
          >
            <p
              className="text-start text-gray-500 "
            >
              Account ID
            </p>
            <div
              className="pl-2 "
            >
              <icon-mock />
            </div>
          </div>
          <div
            className="mb-1.5 border flex flex-row h-10 rounded-lg bg-gray-50 appearance-none border-2 border-gray-300 w-full text-gray-700 leading-tight focus:bg-white focus:border-accent-theme"
          >
            <input
              className="rounded-l-lg bg-gray-50 appearance-none border-0 border-gray-300 w-full py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-accent-theme"
              onChange={[Function]}
              placeholder="yourname"
              type="email"
              value=""
            />
            <div
              className="w-15  h-9 px-2 flex flex-col items-center justify-center border-l-2 border-gray-300"
            >
              .near
            </div>
          </div>
        </div>
        <div
          className="mt-5 "
        >
          <ButtonWithIcon
            isDisabled={true}
            label="Continue"
          >
            <button
              className="bg-silver-sand font-bold py-2 px-4 rounded-lg flex items-center"
              disabled={true}
            >
              <span
                className="mr-3"
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                Continue
              </span>
              <icon-mock />
            </button>
          </ButtonWithIcon>
        </div>
        <div
          className="mt-5 px-5"
        >
          <p
            className="text-center"
          >
            by clicking continue you must agree to near labs
             
            <Link
              href="/"
              passHref={true}
            >
              <span
                href="/"
                onClick={[Function]}
                onMouseEnter={[Function]}
                style={
                  Object {
                    "color": "blue",
                  }
                }
              >
                Terms & Conditions
              </span>
            </Link>
             
            and
             
            <Link
              href="/"
              passHref={true}
            >
              <span
                href="/"
                onClick={[Function]}
                onMouseEnter={[Function]}
                style={
                  Object {
                    "color": "blue",
                  }
                }
              >
                Privacy Policy
              </span>
            </Link>
          </p>
        </div>
        <div
          className="mt-5 w-full px-5"
        >
          <Divder>
            <div
              style={
                Object {
                  "border": "0.5px solid lightgrey",
                }
              }
            />
          </Divder>
        </div>
        <div
          className="mt-5"
        >
          <p
            className="text-center"
          >
            Already have an account?
          </p>
        </div>
        <div
          className="mt-5"
        >
          <ButtonWithIcon
            isDisabled={false}
            isLogin={true}
            label="Log in with NEAR "
          >
            <button
              className="bg-accent-black font-bold py-2 px-4 rounded-lg flex items-center"
              disabled={false}
            >
              <span
                className="mr-3"
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                Log in with NEAR 
              </span>
              <icon-mock />
            </button>
          </ButtonWithIcon>
        </div>
      </div>
    </CreateNEAR>
  </Provider>
</withRedux(CreateNEAR)>
`;
